### PR TITLE
Modified KyberHintHandler to abi.[encode/decode] method

### PR DESCRIPTION
**Pull request by ayobuenavista**

### CODE REVIEW

File: contractsSol5/KyberHintHandler.sol [hunk 1]
```diff
@@ -346,22 +177,139 @@ contract KyberHintHandler is IKyberHint, Utils4 {
...
+        } else {
+            if (splits.length != 0) return HintErrors.SplitsZeroError;
+        }
+
+        return HintErrors.NoError;
+    }
+
+    /// @notice Throws error message to user to indicate error on hint
+    /// @param error Error type from HintErrors enum
+    function throwHintError(HintErrors error) internal pure {
```
Comment (ilanDoron) [inline contractsSol5/KyberHintHandler.sol]: please try a bit shorter message
this is still part of matching engine.
so should be readable
but shorter.

File: contractsSol5/KyberHintHandler.sol [hunk 2]
```diff
@@ -346,22 +177,139 @@ contract KyberHintHandler is IKyberHint, Utils4 {
...
+        internal
+        view
+        returns(
+            TradeType t2eType,
+            IKyberReserve[] memory t2eAddresses,
+            uint[] memory t2eSplits,
+            TradeType e2tType,
+            IKyberReserve[] memory e2tAddresses,
+            uint[] memory e2tSplits,
+            HintErrors verified
```
Comment (ilanDoron) [inline contractsSol5/KyberHintHandler.sol]: valid seems to me a bit more suitable.
not sure though

Comment (ayobuenavista) [reply contractsSol5/KyberHintHandler.sol]: Hmmm okay. Using Valid is fine too.

File: contractsSol5/KyberHintHandler.sol (hunk 1 above)
Comment (ayobuenavista) [reply contractsSol5/KyberHintHandler.sol]: Okay, will try to shorten messages.

**Comment by ayobuenavista** [comment]
Don't merge yet. Adding some missing tests for T2T for duplicate reserveIds. Also fixing some formatting on hintHandler test. :D
